src/uwtools/config/formats/base.py

```diff
@@ -168,33 +199,16 @@ def compare_config(self, dict1: dict, dict2: Optional[dict] = None) -> bool:
         :param dict2: The second dictionary (default: this config).
         :return: True if the configs are identical, False otherwise.
         """
-        dict2 = self.data if dict2 is None else dict2
-        diffs: dict = {}
-
-        for sect, items in dict2.items():
-            for key, val in items.items():
-                if val != dict1.get(sect, {}).get(key, ""):
-                    try:
-                        diffs[sect][key] = f" - {val} + {dict1.get(sect, {}).get(key)}"
-                    except KeyError:
-                        diffs[sect] = {}
-                        diffs[sect][key] = f" - {val} + {dict1.get(sect, {}).get(key)}"
-
-        for sect, items in dict1.items():
-            for key, val in items.items():
-                if val != dict2.get(sect, {}).get(key, "") and diffs.get(sect, {}).get(key) is None:
-                    try:
-                        diffs[sect][key] = f" - {dict2.get(sect, {}).get(key)} + {val}"
-                    except KeyError:
-                        diffs[sect] = {}
-                        diffs[sect][key] = f" - {dict2.get(sect, {}).get(key)} + {val}"
-
-        for sect, keys in diffs.items():
-            for key in keys:
-                msg = f"{sect}: {key:>15}: {keys[key]}"
-                log.info(msg)
-
-        return not diffs
+        dict2 = self.as_dict() if dict2 is None else dict2
+        lines1, lines2 = map(self._compare_config_get_lines, [dict1, dict2])
+        difflines = list(difflib.ndiff(lines2, lines1))
+        if all(line[0] == " " for line in difflines):  # i.e. no +/-/? lines
+            return True
+        if header:
+            self._compare_config_log_header()
+        for diffline in difflines:
+            log.info(diffline.rstrip())
+        return False
```

Comment:
Wow.

Reply:
Concise, right? It helps to reduce a new problem to an existing one with a known solution (useful diff algorithms).